crates/base/src/index.rs

```diff
@@ -101,8 +104,18 @@ pub struct OptimizingOptions {
     #[validate(range(min = 0.01, max = 1.00))]
     pub delete_threshold: f64,
     #[serde(default = "OptimizingOptions::default_optimizing_threads")]
-    #[validate(range(min = 1, max = 65535))]
-    pub optimizing_threads: usize,
+    pub optimizing_threads: AtomicUsize,
```

Comment:
Let's maintain it out of `IndexOptions`. We may create a structure `FlexibleOptions` to save it and use `arc_swap` or something else.